src/control/jobs/XournalScheduler.cpp

```diff
@@ -66,28 +75,21 @@ void XournalScheduler::removeSource(void* source, JobType type, JobPriority prio
     if (awaitFinishTask) {
         finishTask();
     }
-
-    g_mutex_unlock(&this->jobQueueMutex);
 }
 
 auto XournalScheduler::existsSource(void* source, JobType type, JobPriority priority) -> bool {
     bool exists = false;
-    g_mutex_lock(&this->jobQueueMutex);
+    std::lock_guard<std::mutex> lock(this->jobQueueMutex);
```

Comment:
We could use `std::scoped_lock` here (and in similar locations), but I think I agree with this post: https://stackoverflow.com/posts/60172828/timeline#history_38bf9e37-80b9-49e6-ba53-299043622aea.

Reply:
why not using `std::unique_lock`?